### PR TITLE
Weekly openmeteo transform.py: Alex

### LINKED ISSUE #25: Create test file for weekly meteo API transform script
As a developer, I want to test the transform.py file to ensure that the data is being processed correctly and robustly, and that the functions deal with appropriate edge cases and erroneous input effectively so that the user experience is as intended.

### PR DESCRIPTION
**Pull request by AlexOBarnes**
This closes #25 #19 

What I have done:
- Created `transform.py`
- Implemented `transform.py`
- Created `test_transform.py`
- Implemented tests and added conftest fixtures

### CODE REVIEW

File: weekly-openmeteo/test_transform.py [hunk 1]
```diff
@@ -1 +1,80 @@
 '''Contains tests for transform.py'''
+from datetime import datetime as dt
+import pytest
+from transform import format_time, clean_data, format_data, transform
+
+class TestFormatTime:
```
Comment (MurasatoNaoya) [inline weekly-openmeteo/test_transform.py:6]: I like the use of test classes, makes it clear what the overarching goal of a group of tests is.

Comment (AlexOBarnes) [reply weekly-openmeteo/test_transform.py:6]: I agree

File: weekly-openmeteo/transform.py [hunk 2]
```diff
@@ -1 +1,41 @@
...
+
+def clean_data(county_data: dict) -> dict:
+    '''Returns a dictionary of county_id, sunset and sunrise times'''
+    return {'sunrise':county_data['daily']['sunrise'],
+            'sunset':county_data['daily']['sunset']}
+
+def format_data(clean_county_data: dict, county_id: int) -> list[list]:
+    '''Returns a list of lists containing county_id, sunset and sunrise times'''
+    return [[county_id, format_time(clean_county_data['sunrise'][i]),
+            format_time(clean_county_data['sunset'][i])] for i in range(7)]
```
Comment (CalebMathers) [inline weekly-openmeteo/transform.py:17]: Good breakdown of functions to make testing easier

File: weekly-openmeteo/transform.py [hunk 3]
```diff
@@ -1 +1,41 @@
...
+        raise ValueError('No data to transform.')
+
+    formatted_data = []
+    logging.info('Data transformation begun.')
+
+    for county_id,county in enumerate(data,1):
+        logging.info('Cleaning data for county: %s.',county_id)
+        clean_county = clean_data(county)
+        formatted_data.extend(format_data(clean_county,county_id))
+        logging.info('Data cleaning for county %s complete.', county_id)
```
Comment (CalebMathers) [inline weekly-openmeteo/transform.py:31]: Nice precise logging

File: weekly-openmeteo/transform.py [hunk 4]
```diff
@@ -1 +1,41 @@
...
+
+    logging.info('Data tranformation complete.')
+    logging.info('Number of rows to insert: %s',len(formatted_data))
+    return formatted_data
+
+
+if __name__ == '__main__':
+    logger = logging.getLogger(__name__)
+    logging.basicConfig(level=logging.INFO)
+    transform([{}])
```
Comment (MurasatoNaoya) [inline weekly-openmeteo/transform.py:41]: Do we have any measure of coverage?

File: weekly-openmeteo/test_transform.py (hunk 1 above)
Comment (CalebMathers) [inline weekly-openmeteo/test_transform.py:6]: Good to break tests for functions up into classes

File: weekly-openmeteo/conftest.py [hunk 5]
```diff
@@ -38,5 +39,72 @@ def invalid_response_type():
 
 @pytest.fixture
 def valid_coords():
-    """Fixture for invalid API response data."""
+    """Fixture for valid coords."""
```
Comment (CalebMathers) [inline weekly-openmeteo/conftest.py:42]: Fixtures are good, could keep in format with the others and use `'''`s rather than `"""`s here though

File: weekly-openmeteo/test_transform.py [hunk 6]
```diff
@@ -1 +1,80 @@
...
+
+    def test_transform_empty_data(self):
+        '''Tests that empty data raises an error'''
+        with pytest.raises(ValueError):
+            transform([])
+
+    def test_transform_invalid_data(self,invalid_openmeteo_data):
+        '''Tests that invalid data raises and error'''
+        with pytest.raises(IndexError):
+            transform(invalid_openmeteo_data)
```
Comment (MurasatoNaoya) [inline weekly-openmeteo/test_transform.py:80]: All test classes are definielty functional and will test core functionality. For TestCleanData and TestFormatTime, could parametrisation be used to test a wider range of examples? Otherwise LGTM.

**Review by MurasatoNaoya — APPROVED** [review]

File: weekly-openmeteo/conftest.py [hunk 7]
```diff
@@ -38,5 +39,72 @@ def invalid_response_type():
...
+
+@pytest.fixture
+def invalid_openmeteo_data():
+    '''Returns a sample dataset akin to openmeteo'''
+    return [{'daily': {
+            'sunrise': ['2024-10-09T06:00', '2024-10-10T06:01', '2024-10-11T06:02',
+                        '2024-10-12T06:03', '2024-10-13T06:04', '2024-10-14T06:05'],
+            'sunset': ['2024-10-09T18:00', '2024-10-10T18:01', '2024-10-11T18:02',
+                       '2024-10-12T18:03', '2024-10-13T18:04', '2024-10-14T18:05',
+                       '2024-10-15T18:06']}}]*100
```
Comment (CalebMathers) [inline weekly-openmeteo/conftest.py:110]: Nice clean way to create a bulk of test dataset

File: weekly-openmeteo/test_transform.py [hunk 8]
```diff
@@ -1 +1,80 @@
...
+
+    def test_invalid_time_string(self):
+        '''Tests that invalid inputs cause errors'''
+        with pytest.raises(ValueError):
+            format_time('Invalid Time')
+
+    def test_incomplete_time_string(self):
+        '''Tests that missing values cause an error'''
+        with pytest.raises(ValueError):
+            format_time('2024-10-09T15')
```
Comment (CalebMathers) [inline weekly-openmeteo/test_transform.py:20]: Good error testing

**Review by CalebMathers — APPROVED** [review]

File: weekly-openmeteo/transform.py (hunk 4 above)
Comment (AlexOBarnes) [reply weekly-openmeteo/transform.py:41]: I believe all tests files cover at least 75%. Maybe we should incorporate that into the `py_test.yml`
